src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpGoToDefinition.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Common;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests.CSharp
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpGoToDefinition : AbstractEditorTest
+    {
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        public CSharpGoToDefinition(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(CSharpGoToDefinition))
+        {
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.GoToDefinition)]
+        public void GoToClassDeclaration()
+        {
+            AddFile("FileDef.cs");
+            OpenFile(ProjectName, "FileDef.cs");
+            Editor.SetText(
+@"class SomeClass
+{
+}");
+            AddFile("FileConsumer.cs");
+            OpenFile(ProjectName, "FileConsumer.cs");
+            Editor.SetText(
+@"class SomeOtherClass
+{
+    SomeClass sc;
+}");
+            PlaceCaret("SomeClass");
+            Editor.GoToDefinition();
+            VerifyTextContains(@"class SomeClass$$", assertCaretPosition: true);
+            Assert.False(VisualStudio.Instance.Shell.IsActiveTabProvisional());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.GoToDefinition)]
+        public void DoToDefinitionOpensProvisionalTabIfDocumentNotAlreadyOpen()
+        {
+            AddFile("FileDef.cs");
+            OpenFile(ProjectName, "FileDef.cs");
+            Editor.SetText(
+@"class SomeClass
+{
+}");
+            CloseFile(ProjectName, "FileDef.cs");
+            AddFile("FileConsumer.cs");
+            OpenFile(ProjectName, "FileConsumer.cs");
+            Editor.SetText(
+@"class SomeOtherClass
+{
+    SomeClass sc;
+}");
+            PlaceCaret("SomeClass");
+            Editor.GoToDefinition();
+            VerifyTextContains(@"class SomeClass$$", assertCaretPosition: true);
+            Assert.True(VisualStudio.Instance.Shell.IsActiveTabProvisional());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.GoToDefinition)]
+        public void GoToDefinitionWithMultipleResults()
+        {
+            SetUpEditor(
+@"partial class /*Marker*/ $$PartialClass { }
+
+partial class PartialClass { int i = 0; }");
+
+            Editor.GoToDefinition();
+
+            const string programReferencesCaption = "'PartialClass' declarations";
+            var results = VisualStudio.Instance.FindReferencesWindow.GetContents(programReferencesCaption);
+
+            var activeWindowCaption = VisualStudio.Instance.Shell.GetActiveWindowCaption();
+            Assert.Equal(expected: programReferencesCaption, actual: activeWindowCaption);
```

Comment:
Not necessarily. You can imagine that the expected Find References window is open and populated correctly, but not the _active_ window. In that case `GetContents` would succeed but this `Assert.Equal` would fail.